blocks/rich-text/index.js

```diff
@@ -412,14 +395,25 @@ export class RichText extends Component {
 	 *
 	 * @param {boolean} checkIfDirty Check whether the editor is dirty before calling onChange.
 	 */
-	onChange( checkIfDirty = true ) {
-		if ( checkIfDirty && ! this.editor.isDirty() ) {
+
+	onChange() {
+		this.isEmpty = this.editor.dom.isEmpty( this.editor.getBody() );
+		this.savedContent = this.isEmpty ? [] : this.getContent();
+		this.props.onChange( this.savedContent );
+	}
+
+	onAddUndo( { lastLevel } ) {
+		if ( ! lastLevel ) {
```

Comment:
Might help to include a note and/or reference to the fact that TinyMCE creates an initial undo level, which is why it's checked here.

https://github.com/tinymce/tinymce/blob/a4add29/src/core/main/ts/api/UndoManager.ts#L50-L53

Reply:
You remind the that there's actually another even in TinyMCE that steps this... I think in previous iteration it was required to catch the initial bookmark, but now that it's no longer setting focus it should be fine to use the `change` event.

https://github.com/tinymce/tinymce/blob/a4add29/src/core/main/ts/api/UndoManager.ts#L242-L247